### DISCUSSION TITLE
v0 npx: Cannot convert undefined or null to object

### DISCUSSION THREAD

**Discussion by LPTP1Dev** [vercel/community#4732] [Help]
### Summary

After creating some code with v0.dev I wanted to add it using the npx command.
Add command returns:

Configuration is missing. Please run the following command to create a components.json file.

 npx v0@latest init

But when I try to run `npx v0@latest init` I just get a red message saying `Cannot convert undefined or null to object`.
Anyone know how to fix this?

### Steps to Reproduce

Open a new terminal
run `npx v0@latest init`

**Comment by amyegan** [comment]
Thanks for reporting this! We're investigating now.

**Comment by shadcn** [comment]
@LPTP1Dev Hey! I'm looking into this issue. Can you help me reproduce it please? Was it a new Next.js project or an existing one?

**Reply by LPTP1Dev** [reply]
The project the issue occured in was a new Next.js project with typescript support created via WebStorm and yarn. I just slightly edited the index file. Thats basicly it, I just had the Next.js dev server running and executed the add and the init command from v0, just like mentioned.

**Reply by crispyaus** [reply]
also experiencing. fresh nextjs project

**Reply by shadcn** [reply]
Hey @JackCrispy, can you share more details on your setup please? Which Next.js version? You can run `next info` and paste the output here. Thank you.

**Reply by amp127** [reply]
I tried to install on a couple different test environments, sanity blog which is on github failed, but a clean install was fine. 

[v0debug.txt](https://github.com/vercel/community/files/14472016/v0debug.txt)

**Reply by oitent** [reply]
This is happening to me as well…

**Reply by amyegan** [reply]
A missing dependency was the most reliable way I found to replicate the error. Do you already have tailwind installed?

**Reply by AGI-CEO** [reply]
I have the same error and do have tailwind already installed. any updates on this?

**Comment by ngoiyaeric** [comment]
unable to deploy locally getting generic vercel page after running npx run dev

**Reply by amyegan** [reply]
Hey @ngoiyaeric. I wasn't able to replicate this behavior today. If it's still a problem for you, please share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example) in your other post https://github.com/vercel/community/discussions/6580

**Comment by AGI-CEO** [comment]
I have the same problem. I'm using the wordpress CMS starter template and tried updating dependencies too

https://github.com/orgs/vercel/discussions/4732#discussioncomment-9263282

**Comment by shenli** [comment]
I met the same issue. Is there any way to fix it? I upgraded node and npm to the latest version. Still doesn't work.

**Reply by shenli** [reply]
More information:
The issue was met when I ran the command in an existing project https://v0.dev/docs#existing-project
I tried the steps for a new project. It works.

**Comment by noises1990** [comment]
Having the same issue right now, just ran on the image-gallery-starter project already published on vercel:

```
npx v0@latest init
Cannot convert undefined or null to object
```

**Comment by AnthonyFR1** [comment]
look these documentation            https://v0.dev/docs

**Comment by jasonkneen** [comment]
Still a problem. Annoying as I was enjoying using this but now can't benefit from the component library

**Comment by oaVa-o** [comment]
Edit: I got this to work by deleting `node_modules` yet again, then installing `shadcn-ui`, then running `npx shadcn-ui@latest init`, then doing whatever I wanted to do with the `v0` package's CLI.

Note that I'm not sure if installing `shadcn` locally before running the `npx` command changed literally anything at all, nor if I really needed to delete `node_modules`.

Original: This is still very much a problem. I tried using both Bun and NPM, fully deleting `node_modules` multiple times.

**Comment by oitent** [comment]
Just had this issue today, again. After manually creating the component
file I got:
*:::Failed to read project configuration. Make sure your project is a
Next.js project.:::*

Then I decided to start a project from scratch, install V0 and copy the basic configs to the project with the issue:

```JSON
{

  "$schema": "https://ui.shadcn.com/schema.json",

  "style": "default",

  "rsc": true,

  "tsx": true,

  "tailwind": {

    "config": "tailwind.config.ts",

    "css": "app/globals.css",

    "baseColor": "gray",

    "cssVariables": false

  },

  "aliases": {

    "utils": "@/lib/utils",

    "components": "@/components"

  }

}
```


An got this:

*:::v0 now requires the `components.json` file to have `cssVariables` set
to `true`. Please update your `components.json` file.:::*


So, I changed it to true.


Guess what… Problem solved.


PS. At least for me.



I'm guessing that this might have something to do with files/folders
permissions, maybe?


Or the install just can't, won't write the files.

On Wed, Jul 3, 2024 at 2:15 PM oaVa-o ***@***.***> wrote:

> This is still very much a problem. I tried using both Bun and NPM, fully
> deleting node_modules multiple times.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/vercel/discussions/4732#discussioncomment-9950844>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AGVNKCMLXSN5AYTZIUGSY2LZKQWTVAVCNFSM6AAAAAA646IMT2VHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TSNJQHA2DI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*André Renault*
55 31 9 8799 9303

**Comment by shadcn** [comment]
Can confirm the `Cannot convert undefined or null to object` bug. Will look into it.

**Comment by Vaib215** [comment]
any update on this
